Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1

```diff
@@ -414,73 +403,118 @@ function Wait-ServiceFabricApplicationTypeTerminalStatus
         -RetryMessage (Get-VstsLocString -Key SFSDK_RetryingGetApplicationType)
 }
 
-function Unregister-ServiceFabricApplicationTypeAction
+function Wait-ServiceFabricApplicationTypeRegistrationStatus
 {
-    Param(
+    Param (
         [string]
         $ApplicationTypeName,
 
         [string]
         $ApplicationTypeVersion,
 
         [int]
-        $TimeoutSec
+        $TimeoutSec,
+
+        [string]
+        $TimeoutAction,
+
+        [string]
+        $OperationType
     )
 
-    $global:operationId = $SF_Operations.UnregisterApplicationType
+    $global:operationId = $SF_Operations.GetApplicationType
+    $getAppTypeAction = { Get-ServiceFabricApplicationType -ApplicationTypeName $ApplicationTypeName -ApplicationTypeVersion $ApplicationTypeVersion }
+    $getAppTypeRetryEvaluator = {
+        param($appType)
 
-    $unregisterAction = {
-        Unregister-ServiceFabricApplicationType -ApplicationTypeName $ApplicationTypeName -ApplicationTypeVersion $ApplicationTypeVersion -Force -TimeoutSec $TimeoutSec
-        if (!$?)
```

Comment:
this if block should be kept in the Unregister-ServiceFabricApplicationTypeAction's $registerAction delegate

Reply:
Its not being touched as a part of this PR